src/components/AudioPlayer/AudioPlayer.js

```diff
@@ -59,72 +85,123 @@ class AudioPlayerSub extends Component {
   }
 
   checkTrack = async () => {
-    // Check if the track has changed
+    // checkTrack = async (prevProps) => {
+    /*
+    Check if the current track in TrackPlayer (oldTrack) is different
+    than the track in props. This handles the case where a user
+    switches to a screen where the component is already rendered
+    (example: using the back button)
+    */
+    const {
+      topScreen,
+      track,
+      playing,
+      updatePlaying,
+      prevProgress,
+      duration,
+      autoplay,
+    } = this.props
+
     const id = await TrackPlayer.getCurrentTrack()
     const oldTrack = await TrackPlayer.getTrack(id)
-    if (oldTrack?.url != this.props.track.url) {
+
+    if (topScreen && !this.state.switching && oldTrack?.url != track.url) {
+      this.setState({ switching: true })
+
       const id = uuid()
-      const { track, playing, autoplay, updatePlaying } = this.props
       await TrackPlayer.add({
         id,
         url: track.url,
         title: track.title,
         artist: track.subtitle,
         artwork: track.artwork,
       })
-      TrackPlayer.skip(id)
+
+      await TrackPlayer.skip(id)
+
+      //check that new track is ready before playing
+      let isReady = (await TrackPlayer.getState()) == TrackPlayer.STATE_READY
+      while (!isReady) {
+        isReady = (await TrackPlayer.getState()) == TrackPlayer.STATE_READY
+      }
+
+      // reset prevProgress if already
+      // else, revert to previous progress if returning to a previous screen
+      if (Math.round(prevProgress * 100) / 100 === 1) {
+        this.props.updatePrevProgress(0)
+      } else if (prevProgress !== 0) {
+        const { updatePlayed, updateProgress, updatePrevProgress } = this.props
+        const newPlayed = duration * prevProgress
+        updatePlayed(newPlayed)
+        updateProgress(prevProgress)
+        updatePrevProgress(0)
+        const abc = await TrackPlayer.seekTo(newPlayed)
+      }
+
       // If player was already set to play, start playing
       if (playing || autoplay) {
         await TrackPlayer.play()
         updatePlaying(true)
       }
-      let state = await TrackPlayer.getState()
-      if (state == 'playing' && !playing) {
-        const { updatePlaying } = this.props
-        updatePlaying(true)
+
+      this.setState({ switching: false, startSwitch: false })
+    }
+
+    // ensure play/pause button matches playing prop
+    let playerState = await TrackPlayer.getState()
+    if (playerState == TrackPlayer.STATE_PLAYING && !playing) {
+      updatePlaying(true)
+    }
+
+    // clean out unnecessary tracks in TrackPlayer queue
+    await TrackPlayer.getQueue().then(queue => {
+      if (queue.length > 1) {
```

Comment:
Would it be worth removing all of the queued tracks instead of just the first one?

Reply:
Yes! It would! I'll make a new commit with the change

Reply:
Actually, that part of the code is called multiple times so the queue cleans up